src/index.js

```diff
@@ -1,42 +1,79 @@
+let productArray = [];
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  let price = product.querySelector(".price span").innerHTML;
+  let quantity = product.querySelector(".quantity input").value;
+  let currentPrice = Math.floor(price) * Math.floor(quantity);
+  product.querySelector(".subtotal span").innerHTML = currentPrice;
+  return currentPrice;
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+
+  // const singleProduct = document.querySelectorAll(".product");
+  // updateSubtotal(singleProduct);
 
   // ITERATION 2
-  //... your code goes here
+  const multipleProduct = document.querySelectorAll(".product");
+  let finalMultipleProduct = [...multipleProduct];
+  let currentPriceOfEachItem = finalMultipleProduct.map((el) => {
+    return updateSubtotal(el);
+  });
 
   // ITERATION 3
-  //... your code goes here
+  let total = currentPriceOfEachItem.reduce((el, acc) => (acc += el), 0);
+  let totalValue = document.querySelector("#total-value span");
+  totalValue.innerHTML = total;
 }
 
 // ITERATION 4
 
-function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
+function removeProduct(element) {
+  let deleteButton = element.querySelector(".btn-remove");
+  deleteButton.onclick = () => element.parentNode.removeChild(element);
   //... your code goes here
 }
 
 // ITERATION 5
 
-function createProduct() {
-  //... your code goes here
+function createProduct(element) {
```

Comment:
Another way of doing this is by instead of creating a `tr` element and then target that element to give it the class `product`, you could target the `tbody` and insert all the `html` inside:
```
function createProduct() {
  let newProductInputs = document.querySelectorAll(".create-product input");
  let newProductName = newProductInputs[0].value;
  let newProductPrice = newProductInputs[1].value;

  // Saving the newProduct html code in a variable
  let newProduct = `
  <tr class="product">
    <td class="name">
      <span>${newProductName}</span>
    </td>
    <td class="price">$<span>${newProductPrice}</span></td>
    <td class="quantity">
      <input type="number" value="0" min="0" placeholder="Quantity" />
    </td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action">
      <button class="btn btn-remove">Remove</button>
    </td>
  </tr>`;

  // Targeting the table
  const tableBody = document.querySelector("tbody");
  // Adding the newProduct to the end of the table
  tableBody.insertAdjacentHTML("beforeend", newProduct);

  // Adding the removeProduct function to the remove button
  let removeBtn = document.querySelectorAll(".btn-remove");
  removeBtn.forEach((btn) => btn.addEventListener("click", removeProduct));
}
```